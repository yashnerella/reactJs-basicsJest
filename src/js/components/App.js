/**
 * Created by yashw on 24-02-2017.
 */

import React, {Component} from 'react';

export default class App extends Component{
    constructor(){
        super();
        this.state = {
            labelValue: "before Click!"
        }
    }

    _handleButtonClick(){
        if(this.state.labelValue == "before Click!"){
            this.setState({
                labelValue: "after Click!"
            })
        }
        else{
            this.setState({
                labelValue: "before Click!"
            })
        }

    }

    render(){
        return(
            <div>
                <h3>Text Changer</h3>
                <label>{this.state.labelValue}</label><br/>
                <button onClick={this._handleButtonClick.bind(this)}>Change Label Text</button>
            </div>
        );
    }
}