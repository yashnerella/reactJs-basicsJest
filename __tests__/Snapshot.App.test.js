/**
 * Created by yashw on 25-02-2017.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/js/components/App';

describe('App component Snapshot testing',()=>{
    test('snapshots should not match',()=>{
        let component = renderer.create(<App/>);
        let obj = component.toJSON();
        expect(obj).toMatchSnapshot();
        component = renderer.create(<App/>);
         obj = component.toJSON();
        expect(obj).toMatchSnapshot();
    })
});
