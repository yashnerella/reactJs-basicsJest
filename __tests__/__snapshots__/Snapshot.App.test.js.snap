// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Snapshot testing snapshots should not match 1`] = `
<div>
  <h3>
    Text Changer
  </h3>
  <label>
    before Click!
  </label>
  <br />
  <button
    onClick={[Function]}
  >
    Change Label Text
  </button>
</div>
`;

exports[`App component Snapshot testing snapshots should not match 2`] = `
<div>
  <h3>
    Text Changer
  </h3>
  <label>
    before Click!
  </label>
  <br />
  <button
    onClick={[Function]}
  >
    Change Label Text
  </button>
</div>
`;
